src/out-of-bounds-access.c

```diff
@@ -0,0 +1,59 @@
+#include <stdio.h>
+#include <assert.h>
+#include <string.h>
+#include <stdlib.h>
+
+// An example of a user changing the root password by entering a username that's long enough (>10 characters) to go out of bounds of the allocated input array.
+// The password is replaced with the out of bounds characters e.g. if `root------123` is inputted for a username, the password is now `123`.
+
+int checkInput(char input[], char toMatch[])
+{
+    input[strcspn(input, "\n")] = '\0'; // strip \n from end of fgets
+    if (strcmp(input, toMatch))
+    {
```

Comment:
Nothing against, but why this formatting?

Reply:
`{` on new line seems to be the convention (looking at other c files in the repo). My formatter did it for me automatically